File: JavaScript/293-Flip Game.js
/**
 * @param {string} s
 * @return {string[]}
 */
var generatePossibleNextMoves = function(s) {
    let res = [];
    
    for (let i=0; i<s.length; i++){
        if (s.substring(i,i+2) === '++'){
            res.push(s.substring(0,i).concat("--").concat(s.substring(i+2)));
        }
    }
    return res;
};

File: C++/293-Flip Game.cpp
class Solution {
public:
    vector<string> generatePossibleNextMoves(string currentState) {
        vector<string> res;
        for (int i=1; i<currentState.size(); i++) {
            if(currentState[i] == '+' && currentState[i-1] == '+') {
                string toPush = currentState;
                toPush[i] = '-';
                toPush[i-1] = '-';
                res.push_back(toPush);
            }
        }
        return res;
    }
};